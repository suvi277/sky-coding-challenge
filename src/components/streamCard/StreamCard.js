import React from 'react';
import './StreamCard.scss';

const StreamCard = (props) => (
	<div className="card shadow-lg">
		<div className="stream-media">
			<img className="card-img-top img-fluid" src={props.assetImage} alt={props.name} />
		</div>
		<div className="card-title pt-3">
			<p className="title float-left w-75">{props.name}</p>
			<p className="views float-right">
				<span className="fa fa-eye" />
				<span className="pl-2 view-number">{props.skygoTotalViews}</span>
			</p>
		</div>
		<div className="card-overlay">
			<p className="provider">
				<strong>Provider</strong> : <span>{props.provider}</span>
			</p>
			<p className="desc">{props.description}</p>
		</div>
	</div>
);

export default StreamCard;
