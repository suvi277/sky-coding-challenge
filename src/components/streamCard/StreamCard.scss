@import '../../_shared/styles/variable';
@import '../../_shared/styles/mixin';

.card {
  margin: 20px 15px;
  padding: 10px;
  border: none;
  background-color: $white;
  cursor: pointer;
  transition: transform 1s ease;

  .desc {
    font-size: 12px;
  }
  .card-title {
    font-size: 12px;
    font-weight: 800;
    color: $success;

    p {
      margin: 0;
    }
  }

  .card-overlay {
    .provider {
      font-size: 14px;
      margin-bottom: 10px;
    }

    .desc {
      font-size: 12px;
    }
  }

  .stream-media img {
    transition: transform 1.5s ease;
    will-change: transform;
  }

  &:hover {
    @include transform(scale(1.1));

    background-color: $info;
    color: $white;

    .card-title {
      color: $light;
    }

    .stream-media img {
      @include transform(scale(1.05));
    }
  }

  @media screen and (min-width: 576px) {
    width: calc(100% / 2 - 30px);
  }

  @media screen and (min-width: 768px) {
    width: calc(100% / 4 - 30px);
  }
}
