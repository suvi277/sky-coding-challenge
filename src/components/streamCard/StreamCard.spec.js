import React from 'react';
import { shallow, mount } from 'enzyme';
import StreamCard from './StreamCard';

describe('StreamCard Component', () => {
	const wrapper = shallow(<StreamCard />);
	it('renders as expected', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('should exist card and shadow-lg classname', () => {
		expect(wrapper.find('.card.shadow-lg')).toHaveLength(1);
	});

	it('should insert the prop values to correct places ', () => {
		const props = {
				assetImage: 'Some Asset Image',
				name: 'Some Name',
				skygoTotalViews: 100,
				provider: 'Some provider',
				description: 'Some Description'
			},
			streamCardComp = mount(<StreamCard {...props} />);

		expect(streamCardComp.find('.title').render().text()).toEqual('Some Name');
		expect(streamCardComp.find('.card-img-top').render().attr('src')).toEqual('Some Asset Image');
		expect(streamCardComp.find('.view-number').render().text()).toEqual('100');
		expect(streamCardComp.find('.provider span').render().text()).toEqual('Some provider');
		expect(streamCardComp.find('.desc').render().text()).toEqual('Some Description');
	});
});
