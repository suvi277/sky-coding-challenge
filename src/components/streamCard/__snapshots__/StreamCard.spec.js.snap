// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StreamCard Component renders as expected 1`] = `
<div
  className="card shadow-lg"
>
  <div
    className="stream-media"
  >
    <img
      className="card-img-top img-fluid"
    />
  </div>
  <div
    className="card-title pt-3"
  >
    <p
      className="title float-left w-75"
    />
    <p
      className="views float-right"
    >
      <span
        className="fa fa-eye"
      />
      <span
        className="pl-2 view-number"
      />
    </p>
  </div>
  <div
    className="card-overlay"
  >
    <p
      className="provider"
    >
      <strong>
        Provider
      </strong>
       : 
      <span />
    </p>
    <p
      className="desc"
    />
  </div>
</div>
`;
